File: star-framework-module/star-framework-module-order/src/main/java/com/star/truffle/module/order/domain/Order.java
/**create by framework at 2018年09月21日 15:21:35**/
package com.star.truffle.module.order.domain;

import lombok.Getter;
import lombok.Setter;
import java.util.Date;

@Getter
@Setter
public class Order {

  // 主键
  private Long orderId;
  // 订单编号
  private Long orderCode;
  // 订单类型 1自主下单 2代客下单
  private Integer type;
  // 用户ID
  private Long memberId;
  // openid
  private String openId;
  // 用户姓名
  private String name;
  // 用户手机号
  private String mobile;
  // 收货类型 1自提 2快递
  private Integer deliveryType;
  // 收货地址-省
  private String provinceName;
  // 收货地址-市
  private String cityName;
  // 收货地址-区县
  private String areaName;
  // 收货地址
  private String deliveryAddress;
  // 收件人
  private String deliveryName;
  // 收件人手机
  private String deliveryMobile;
  // 快递费 单位分
  private Integer despatchMoney; 
  // 总金额 不包含运费
  private Integer totalMoney;
  // 直属分销商总提成
  private Integer totalBrokerage;
  // 上级分销商总提成
  private Integer totalBrokerageFirst;
  // 分销区域
  private Long regionId;
  // 分销商
  private Long distributorId;
  // 自提地址
  private String shopAddress;
  // 店铺名称
  private String shopName;
  // 店铺电话
  private String shopMobile;
  // 订单状态 1待付款 2待提货 3已提货 4已退货
  private Integer state;
  // 运输状态 1待发货 2已发货 3已完成
  private Integer transportState;
  // 是否删除 1是 0否
  private Integer deleted;
  // 提货码
  private String pickupCode;
  // 创建日期
  private Date createTime;
  // 1用户自己 2系统 3后台
  private Integer deleteUser;
  // 备注
  private String remark;
  // 快递单号
  private String expressNumber;
}

File: blyx-h5/WebContent/js/bl/order.js
var orderId = getParam("orderId");
mui('.main').on('tap', '.cancel', function() {
  mui('.mui-popover').popover('hide');
});
mui('.mui-scroll-wrapper').scroll({
  deceleration: 0.0005
});
$(function() {
  initOrderDetail();
});
var states={'11': '待支付', '12': '待提货', '13': '已提货', '14': '已退货', '15': '已删除', '21': '待支付', '22': '待发货', '23': '已收货', '24': '已退货', '25': '已删除'};
var btnnames = {1: '去支付', 2: '确认提货', 3: '删除', 4: '删除', 5: ''};
function initOrderDetail(){
  var originalPrice = 0;
  var totalMoney = 0;
  var num = 0;
  ajax({
    url: '/api/order/getOrder',
    data: {'orderId' : orderId},
    success: function(data){
      $(".goodslist ul").html("");
      var items = data.details;
      if(null != items && items.length > 0){
        for(var o in items){
          var item = items[o];
          originalPrice += item.originalPrice / 100.0 * item.count;
          totalMoney += item.price / 100.0 * item.count;
          num += item.count;
          $(".goodslist ul").append('<li>\
              <span><img src="'+IMAGE_PREFIX+item.mainPictureUrl+'" alt=""></span>\
              <p>'+item.title+'</p>\
              <p>'+item.specification+'</p>' +
              '<p class="red">' + (data.deliveryType==2 ? '&nbsp;' : (item.pickupTime != null ? item.pickupTime.formatDate('M月d日 h点') : '随时')+'可提货') + '</p>' +
              '<p class="price">￥<b>'+((item.price / 100.0).toFixed(2))+'</b>\
                <del>$'+((item.originalPrice / 100.0).toFixed(2))+'</del>\
                <span class="fl">x'+item.count+'</span>\
              </p>\
            </li>');
        }
      }
      $(".orderstate").text(states[data.deliveryType + '' + data.state]);
      $(".totaldiv .productNum").text(num);
      $(".totaldiv .productMoney").text("￥"+totalMoney.toFixed(2));
      $(".totaldiv .productOriginalPrice").text("￥"+originalPrice.toFixed(2));
      if(data.deliveryType==2){
        $(".totaldiv .despatchMoney").text("￥"+(data.despatchMoney/100.0).toFixed(2));
        $("#item2").append('<p>收件人：'+ data.deliveryName + '</p>');
        $("#item2").append('<p>收件人电话：'+ data.deliveryMobile + '</p>');
        $("#item2").append('<p>收件人地址：'+ data.provinceName + data.cityName + data.areaName + data.deliveryAddress + '</p>');
      }else{
        $(".totaldiv .despatchMoney").parent().remove();
        $("#item2").append('<p>取件人：'+ data.name + '</p>');
        $("#item2").append('<p>取件电话：'+ data.mobile + '</p>');
        $("#item2").append('<p>取件地址：'+ data.regionProvinceName + data.regionCityName + data.regionAreaName + data.shopAddress + '</p>');
        $("#item2").append('<p>店铺名称：'+ data.shopName + '</p>');
      }
      $("#item2").removeClass("mui-hidden");
      $(".totaldiv .productTotalMoney").text("￥"+totalMoney.toFixed(2));
      var dp = data.discountedPrice ? data.discountedPrice : 0;
      $(".totaldiv .discountedPrice").text("￥"+(dp/100.0).toFixed(2));
      totalMoney -= dp/100.0;
      $(".totaldiv .productTotalMoney2").text("￥"+totalMoney.toFixed(2));
      $(".ordercode span").text(data.orderCode);
      $(".ordertime span").text(data.createTime);
    }
  });
}